update test_vulkan

tktrungna opened the pull request:
Fixes #62380

* update test functions to call wheel install folder {sitepackages}/torch instead of build/ folder
* add symbolic link for shared libraries which are called by the tests (this is a bit hacky and should be fixed the rpath before compiling -- similar to https://github.com/pytorch/pytorch/blob/master/.jenkins/pytorch/test.sh#L204-L208).

### Test plan
check if all ci workflows pass

Stack from [ghstack](https://github.com/ezyang/ghstack):
* #62838
* #62797
* #62796
* __->__ #62795
* #62781

Differential Revision: [D30124421](https://our.internmc.facebook.com/intern/diff/D30124421)

tktrungna committed:
update test_vulkan

[ghstack-poisoned]

tktrungna committed:
Update on "update test_vulkan"




[ghstack-poisoned]

tktrungna commented:
@tktrungna has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30124421).

tktrungna committed:
Update on "update test_vulkan"


Differential Revision: [D30124421](https://our.internmc.facebook.com/intern/diff/D30124421)

[ghstack-poisoned]

tktrungna committed:
Update on "update test_vulkan"


Differential Revision: [D30124421](https://our.internmc.facebook.com/intern/diff/D30124421)

[ghstack-poisoned]

tktrungna committed:
Update on "update test_vulkan"


Differential Revision: [D30124421](https://our.internmc.facebook.com/intern/diff/D30124421)

[ghstack-poisoned]

tktrungna committed:
Update on "update test_vulkan"


Differential Revision: [D30124421](https://our.internmc.facebook.com/intern/diff/D30124421)

[ghstack-poisoned]

tktrungna commented:
@tktrungna has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30124421).

tktrungna committed:
Update on "update test_vulkan"

Fixes #62380

* update test functions to call wheel install folder {sitepackages}/torch instead of build/ folder
* add symbolic link for shared libraries which are called by the tests (this is a bit hacky and should be fixed the rpath before compiling -- similar to https://github.com/pytorch/pytorch/blob/master/.jenkins/pytorch/test.sh#L204-L208).

### Test plan
check if all ci workflows pass


Differential Revision: [D30124421](https://our.internmc.facebook.com/intern/diff/D30124421)

[ghstack-poisoned]

tktrungna commented:
@tktrungna has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30124421).

tktrungna committed:
Update on "update test_vulkan"

Fixes #62380

* update test functions to call wheel install folder {sitepackages}/torch instead of build/ folder
* add symbolic link for shared libraries which are called by the tests (this is a bit hacky and should be fixed the rpath before compiling -- similar to https://github.com/pytorch/pytorch/blob/master/.jenkins/pytorch/test.sh#L204-L208).

### Test plan
check if all ci workflows pass


Differential Revision: [D30124421](https://our.internmc.facebook.com/intern/diff/D30124421)

[ghstack-poisoned]

tktrungna committed:
Update on "update test_vulkan"

Fixes #62380

* update test functions to call wheel install folder {sitepackages}/torch instead of build/ folder
* add symbolic link for shared libraries which are called by the tests (this is a bit hacky and should be fixed the rpath before compiling -- similar to https://github.com/pytorch/pytorch/blob/master/.jenkins/pytorch/test.sh#L204-L208).

### Test plan
check if all ci workflows pass


Differential Revision: [D30124421](https://our.internmc.facebook.com/intern/diff/D30124421)

[ghstack-poisoned]

zhouzhuojie commented on .jenkins/pytorch/test.sh:
```diff
@@ -278,12 +278,14 @@ test_libtorch() {
 
 test_vulkan() {
   if [[ "$BUILD_ENVIRONMENT" == *vulkan-linux* ]]; then
+    ln -sf "$TORCH_LIB_DIR"/libtorch* "$TORCH_TEST_DIR"
+    ln -sf "$TORCH_LIB_DIR"/libc10* "$TORCH_TEST_DIR"
```
what's the source of getting to know which `lib*` to `ln` to the TEST_DIR?

tktrungna replied:
there maybe better approach but what I am doing now is reading ci failure logs

walterddr replied:
I was wondering if we can simply just symlink everything?

tktrungna replied:
I thought about that but believe there should be a better way with cmake to let executable file looks for *.so files in the `lib` folder, so just leave to command here in this way to let it more trackable. Will create an issue to follow up soon

tktrungna commented:
@tktrungna has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30124421).

tktrungna committed:
Update on "update test_vulkan"

Fixes #62380

* update test functions to call wheel install folder {sitepackages}/torch instead of build/ folder
* add symbolic link for shared libraries which are called by the tests (this is a bit hacky and should be fixed the rpath before compiling -- similar to https://github.com/pytorch/pytorch/blob/master/.jenkins/pytorch/test.sh#L204-L208).

### Test plan
check if all ci workflows pass


Differential Revision: [D30124421](https://our.internmc.facebook.com/intern/diff/D30124421)

[ghstack-poisoned]

tktrungna commented:
@tktrungna has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30124421).
